src/Sylius/Bundle/UserBundle/Validator/Constraints/NotRegisteredUser.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UserBundle\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+class NotRegisteredUser extends Constraint
+{
+    public $message = 'This email is not registered. Please register first.';
```

Comment:
What do you think about adding here translation key instead of hard coded message ?